alibi/explainers/partial_dependence.py

```diff
@@ -0,0 +1,1294 @@
+        -------
+        `Explanation` object.
+        """
+        feature_deciles, feature_values = [], []
+        pd_values = [] if kind in [Kind.AVERAGE, Kind.BOTH] else None  # type: Optional[List[np.ndarray]]
+        ice_values = [] if kind in [Kind.INDIVIDUAL, Kind.BOTH] else None  # type: Optional[List[np.ndarray]]
+
+        for pd in pds:
+            feature_values.append(pd['values'])
+            feature_deciles.append(pd['deciles'])
+
+            if (pd_values is not None) and (Kind.AVERAGE in pd):
+                pd_values.append(pd[Kind.AVERAGE])
+            if (ice_values is not None) and Kind.INDIVIDUAL in pd:
+                ice_values.append(pd[Kind.INDIVIDUAL])
+
+        data = copy.deepcopy(DEFAULT_DATA_PD)
+        data.update(
+            feature_names=feature_names,
+            feature_values=feature_values,
+            ice_values=ice_values,
+            pd_values=pd_values,
+            feature_deciles=feature_deciles,
+        )
+        data['meta'].update(
+            feature_names=self.feature_names,
+            categorical_names=self.categorical_names,
+            target_names=self.target_names,
+            response_method=response_method,
+            method=method,
+            kind=kind
+        )
+        return Explanation(meta=copy.deepcopy(self.meta), data=data)
+
+    def reset_predictor(self, predictor: Any) -> None:
+        """
+        Resets the predictor function
+
+        Parameters
+        ----------
+        predictor
+            New `sklearn` estimator.
+        """
+        self.predictor = predictor
+
+
+# No type check due to the generic explanation object
+@no_type_check
+def plot_pd(exp: Explanation,
+            features_list: Union[List[int], Literal['all']] = 'all',
+            target_idx: int = 0,
+            n_cols: int = 3,
+            n_ice: Union[str, int, List[int]] = 'all',
+            centered: bool = True,
+            levels: int = 8,
+            ax: Optional[Union['plt.Axes', np.ndarray]] = None,
+            sharey: str = 'all',
+            pd_num_kw: Optional[dict] = None,
+            ice_num_kw: Optional[dict] = None,
+            pd_cat_kw: Optional[dict] = None,
+            ice_cat_kw: Optional[dict] = None,
+            pd_num_num_kw: Optional[dict] = None,
+            pd_num_cat_kw: Optional[dict] = None,
+            pd_cat_cat_kw: Optional[dict] = None,
+            fig_kw: Optional[dict] = None,
+            seed: Optional[int] = None) -> 'np.ndarray':
+    """
+    Plot partial dependence curves on matplotlib axes.
+
+    Parameters
+    ----------
+    exp
+        An `Explanation` object produced by a call to the
+        :py:meth:`alibi.explainers.partial_dependence.PartialDependence.explain` method.
+    features_list
+        A list of features for which to plot the partial dependence curves or ``'all'`` for all features.
+        Can be an integers denoting feature index denoting entries in `exp.feature_names`. Defaults to ``'all'``.
```

Comment:
Sentence needs rewriting. Also probably need to clarify that the indexing also includes the 2-way tuples.